libcxx/src/ryu/d2s.cpp

```diff
@@ -479,7 +479,7 @@ struct __floating_decimal_64 {
           36893488u, 7378697u, 1475739u, 295147u, 59029u, 11805u, 2361u, 472u, 94u, 18u, 3u };
 
         unsigned long _Trailing_zero_bits;
-#if _LIBCPP_HAS_BITSCAN64
+#if !defined(_MSC_VER) || defined(_M_AMD64) || defined(_M_ARM64) // we have own _BitScanForward64 for non-MSVC
```

Comment:
Can't we just use `std::countr_zero`? That'd make the code below unnecessary and we can drop this condition altogether.

Reply:
std::to_chars is C++17, std::countr_zero is C++20, but probably std::__countr_zero will work in C++17 mode

codegen is different for x64 https://godbolt.org/z/TjeT661eT, countr_zero translates to tzcnt == rep bsf, that properly works with zero input, but may be more expensive - but benefits of the standard function can be more important

Reply:
You're comparing apples and oranges here. When using clang the code gen is almost identical, with just a single additional mov. I doubt very much that a single mov makes a significant difference in performance. If you can show a significant difference in performance we can also use `__builtin_ctzg` directly with a comment. Re. C++20: we're in dylib code, so when a feature was introduced doesn't matter.